moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp

```diff
@@ -0,0 +1,351 @@
+std::vector<Bounds> positionConstraintMsgToBoundVector(const moveit_msgs::PositionConstraint& pos_con)
+{
+  auto dims = pos_con.constraint_region.primitives.at(0).dimensions;
+
+  // dimension of -1 signifies unconstrained parameter, so set to infinity
+  for (auto& dim : dims)
+  {
+    if (dim == -1)
+      dim = std::numeric_limits<double>::infinity();
+  }
+
+  return { { -dims[0] / 2, dims[0] / 2 }, { -dims[1] / 2, dims[1] / 2 }, { -dims[2] / 2, dims[2] / 2 } };
+}
+
+std::vector<Bounds> orientationConstraintMsgToBoundVector(const moveit_msgs::OrientationConstraint& ori_con)
+{
+  std::vector<double> dims{ ori_con.absolute_x_axis_tolerance, ori_con.absolute_y_axis_tolerance,
+                            ori_con.absolute_z_axis_tolerance };
+
+  // dimension of -1 signifies unconstrained parameter, so set to infinity
+  for (auto& dim : dims)
+  {
+    if (dim == -1)
+      dim = std::numeric_limits<double>::infinity();
+  }
+  return { { -dims[0] / 2, dims[0] / 2 }, { -dims[1] / 2, dims[1] / 2 }, { -dims[2] / 2, dims[2] / 2 } };
+}
+
+/******************************************
+ * Constraint Factory
+ * ****************************************/
+
+std::shared_ptr<BaseConstraint> createConstraint(robot_model::RobotModelConstPtr robot_model, const std::string& group,
+                                                 const moveit_msgs::Constraints constraints)
+{
+  std::size_t num_dofs = robot_model->getJointModelGroup(group)->getVariableCount();
+  std::size_t num_pos_con = constraints.position_constraints.size();
+  std::size_t num_ori_con = constraints.orientation_constraints.size();
+
+  if (num_pos_con > 0 && num_ori_con > 0)
+  {
+    ROS_WARN_NAMED(LOGNAME, "Combining position and orientation constraints will ignore the z orientation tolerance.");
+    auto con = std::make_shared<PoseConstraint>(robot_model, group, num_dofs);
+    con->init(constraints);
+    return con;
+  }
+  else if (num_pos_con > 0)
+  {
+    if (num_pos_con > 1)
+    {
+      ROS_WARN_NAMED(LOGNAME, "Only a single position constraints supported. Using the first one.");
+    }
+    auto pos_con = std::make_shared<PositionConstraint>(robot_model, group, num_dofs);
+    // auto pos_con = std::make_shared<XPositionConstraint>(robot_model, group, num_dofs);
+    pos_con->init(constraints);
+    return pos_con;
+  }
+  else if (num_ori_con > 0)
+  {
+    if (num_ori_con > 1)
+    {
+      ROS_WARN_NAMED(LOGNAME, "Only a single orientation constraints supported. Using the first one.");
+    }
+
+    auto ori_con = std::make_shared<AngleAxisConstraint>(robot_model, group, num_dofs);
+    ori_con->init(constraints);
+    return ori_con;
+  }
+  else
+  {
+    ROS_ERROR_NAMED(LOGNAME, "No path constraints found in planning request.");
+    return nullptr;
+  }
+}
+
+/******************************************
+ * Angular velocity conversion
+ * ****************************************/
+Eigen::Matrix3d angularVelocityToAngleAxis(double angle, const Eigen::Vector3d& axis)
```

Comment:
@mamoll I suspect this is another bottleneck, runtime wise. (Similar to the one discussed in the meeting yesterday in `calcError`.) But I did not measure it. Based on running some examples and a rough estimate, it seemed the have a similar runtime to using the default numerical Jacobian for the constraints.